experimental/logical-assignment-operators.md

```diff
@@ -0,0 +1,30 @@
+# [Logical Assignment Operators][proposal-logical-assignment]
+
+## Expressions
+
+### LogicalAssignmentExpression
+
+```js
+interface LogicalAssignmentExpression <: Expression {
```

Comment:
Why do we need a new node type for this rather than extending `AssignmentOperator` in [`AssignmentExpression`](https://github.com/estree/estree/blob/master/es5.md#assignmentexpression)?

Reply:
Ah, sorry, I see it's explained below.

Reply:
Just to make sure I understand this: a new node type is necessary because logical assignment expressions are short-circuited?

Reply:
As far as I understood, it's for symmetry with `BinaryExpression` vs `LogicalExpression` (which were initially separate due to short-circuiting, I suppose).

Personally, I'd rather have former to be a single node type as well, but that ship has sailed.

As for new node types, I don't have a strong opinion and fine with either a new node type or reusing `AssignmentExpression`. Would like to hear opinions of others.

Reply:
Yes, it's for symmetry with `BinaryExpression` vs `LogicalExpression`, and I thought that's because of short-circuit.

Reply:
a question here from my relative lack of knowledge of the history of ESTree, and of how ASTs should be designed.

Why is it that ESTree wants to encode this runtime behaviour in the AST?
Isn't the short-circuiting behaviour semantic meaning derived from the operator?

Is there tangible value for consumers of the spec with this distinction?

I ask because, for example, TypeScript's AST representation just uses a single `BinaryExpression` node for both cases.

Reply:
@bradzacher Yeah, as I said above

> Personally, I'd rather have former to be a single node type as well, but that ship has sailed.

This has been there from the beginning of SpiderMonkey AST and I can only assume that it was for encoding of short-circuiting behaviour, but personally I'd also just use a single node type and encourage consumers to check the operator.

Not sure whether to follow the precedent or do something simpler for the new node types here. *Side-note: this is why I said I'm very cautious of accidentally setting new precedents in #204 - they can affect design of nodes far away from the original addition.*

Reply:
I don't have any strong opinion about the choice of following the precedent or using `AssignmentExpression` simply.

Reply:
This is why I asked, as I’m not sure if we want to encode short-circuiting operations separately. My preference would be to reuse AssignmentExpression. If the intent is to execute based on the AST, you’ll always need to look at the operator anyway, and that seems like the logical time to apply any short circuiting behavior.

Reply:
I'm OK with either, but I hope consistent with #204. If we decide to not make new nodes for short-circuiting, I think it's better if #204 is consistent with that.

Reply:
I have updated this PR to extending `AssignmentOperator` because the separation of nodes by short-circuiting looks not important to ESTree.